kernel/src/expressions/column_names.rs

```diff
@@ -204,43 +248,63 @@ impl std::str::FromStr for ColumnName {
     type Err = Error;
 
     fn from_str(s: &str) -> DeltaResult<Self> {
-        // Ambiguous case: The empty string `""`could reasonably parse as either `ColumnName::new([""])`
-        // or `ColumnName::new([])`. However, `ColumnName::new([""]).to_string()` is `"[]"` and
-        // `ColumnName::new([]).to_string()` is `""`, so we choose the latter because it produces a
-        // lossless round trip from `ColumnName` to  `String` and back.
-        let mut chars = s.chars().peekable();
-        drop_leading_whitespace(&mut chars);
-        if chars.peek().is_none() {
-            return Ok(ColumnName::new(&[] as &[String]));
+        let (col, ending) = parse_column_name(&mut s.chars().peekable())?;
+        if ending == FieldEnding::NextColumn {
+            Err(Error::generic("Trailing comma in column name"))
+        } else {
+            Ok(col)
         }
+    }
+}
 
-        let mut path = vec![];
-        loop {
-            drop_leading_whitespace(&mut chars);
-            let (field_name, done) = if chars.next_if_eq(&'`').is_none() {
-                parse_simple(&mut chars)?
-            } else {
-                parse_escaped(&mut chars)?
-            };
-            path.push(field_name);
-            if done {
-                return Ok(ColumnName::new(path));
-            }
+type Chars<'a> = Peekable<std::str::Chars<'a>>;
+
+fn parse_column_name(chars: &mut Chars<'_>) -> DeltaResult<(ColumnName, FieldEnding)> {
+    // Ambiguous case: The empty string `""`could reasonably parse as either `ColumnName::new([""])`
+    // or `ColumnName::new([])`. However, `ColumnName::new([""]).to_string()` is `"[]"` and
+    // `ColumnName::new([]).to_string()` is `""`, so we choose the latter because it produces a
+    // lossless round trip from `ColumnName` to `String` and back. We also swallow a leading comma
+    // to produce an empty column, so that the string "," parses as two empty columns.
+    drop_leading_whitespace(chars);
+    if chars.peek().is_none() {
+        return Ok((ColumnName::empty(), FieldEnding::InputExhausted));
+    }
+    if chars.next_if_eq(&',').is_some() {
```

Comment:
should we make the delimiter a parameter?

Reply:
I really hope we don't ever need to support configurable delimiters...

Reply:
hahaha yea fair enough

Reply:
Actually tho... naming these magic constants is probably wise. They just don't need to be "parameters" in the "choosable at runtime" sense. I'll do that as a follow-up PR since it affects all the column name code, not just this new part.